### PR TITLE
Clone a list and random pointers

### PR DESCRIPTION
**Pull request by arzitmahajan**
## Related Issue
#1042 

## Proposed Changes
1.Code of clone a list with comments added 
2.Separate readme with explaination

## Additional Info
     Code is checked before submitting it is a hard level question and is asked frequently in many in interview rounds
## Checklist

- [x ] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| Input 1 (Output is displayed with debugger)|
| 
image
 | Input 2|
 | 
image
 |

### CODE REVIEW

**Review by Lakhankumawat — COMMENTED** [review]
Your code looks a bit hardcoded in main

**Review by Lakhankumawat — APPROVED** [review]
README looks good ✨

**Comment by arzitmahajan** [comment]
@Lakhankumawat is this PR good if not give me some suggestion how to improve it?

File: L-LinkedList/Clone a List with Random pointers/linked list clone.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	//Basic list
+	Node* root = new Node(1);
+	Node* second = new Node(2);
+	Node* third = new Node(3);
+	Node* fourth = new Node(4);
+	Node* fifth = NULL;
+
+	root->next = second;
+	root->rand = fourth;
+	second->next = third;
```
[the remark this answers is not in the document]
Comment (arzitmahajan) [reply L-LinkedList/Clone a List with Random pointers/linked list clone.cpp:87]: Okay will change this

**Comment by siddhi-244** [comment]
> @siddhi-244 Please review now the function is added

No you haven't taken the input!

File: L-LinkedList/Clone a List with Random pointers/linked list clone.cpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+	cout << "Enter the total number of nodes " << endl<<"->";
+	int n;
+	cin >>n;
+	while (n--) {
+		int val;
+		cout << "Enter node's value" << endl << "->";
+		cin >> val;
+		append(root, val);
+
```
Comment (siddhi-244) [inline L-LinkedList/Clone a List with Random pointers/linked list clone.cpp:106]: Where Iis the closing bracket ?

**Comment by arzitmahajan** [comment]
Sorry I thought separate function is needed, I have made the changes please review now

File: L-LinkedList/Clone a List with Random pointers/linked list clone.cpp (hunk 2 above)
Comment (arzitmahajan) [reply L-LinkedList/Clone a List with Random pointers/linked list clone.cpp:109]: Sorry my mistake

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
Can you give a more detailed approach or algorithm?

**Comment by arzitmahajan** [comment]
> Can you give a more detailed approach or algorithm?

I have Attached debugged pics and Approach in read-me and I think that's all I can do to explain the code .I know it's a bit confusing as this is a hard level question.

**Review by siddhi-244 — APPROVED** [review]
LGTM

**Comment by Lakhankumawat** [comment]
Follow the announcements in discord , why did you modified A-Array's README

**Comment by arzitmahajan** [comment]
> Follow the announcements in discord , why did you modified A-Array's README

Actually I  had contributed an array based question and  while I was updating the readme for Clone a list. I thought it would be better that whenever user goes through the readme and see the heading he can simply click on the heading to go to the code page , so I updated it by adding the link.

**Comment by Lakhankumawat** [comment]
image

**Comment by arzitmahajan** [comment]
done

**Review by Lakhankumawat — APPROVED** [review]
## LGTM 🎉✨

**Pull request merged**